doc/source/examples/sharp-immersed-boundary-solver/3d-composite-mixer-with-pbt-impeller/3d-composite-mixer-with-pbt-impeller.rst

```diff
@@ -0,0 +1,155 @@
+3D Mixer with pitched-blade turbine impeller using Composite Sharp-immersed boundary
+=====================================================================================
+
+The mixing of stirred-tanks is a common chemical engineering problem that can be tackled through immersed boundary simulation. This example, a variation of :doc:`../../incompressible-flow/3d-nitsche-mixer-with-pbt-impeller/nitsche-mixer-with-pbt-impeller`, illustrates how the transient flow in a stirred-tank can be simulated by Lethe using the Sharp-Immersed Boundary formulation with a composite shape.
+
+Features
+----------------------------------
+- Solvers: ``gls_sharp_navier_stokes_3d``
+- Transient problem
+- Rotating complex solid  modelled with the sharp immersed boundary method
+
+Files used in this example
+----------------------------
+
+* Parameter file: ``/examples/sharp-immersed-boundary-solver/3d-composite-mixer-with-pbt-impeller/mixer.prm``
+* Composite geometry file: ``/examples/sharp-immersed-boundary-solver/3d-composite-mixer-with-pbt-impeller/impeller.composite``
+
+Description of the case
+-----------------------
+
+In this example, we simulate a mixer using a PBT impeller through the usage of Sharp-Immersed boudaries. The shape of the impeller is defined by a composition of basic shapes.
+
+Creation of the composite shape file
+------------------------------------
+
+The shape is created in two steps inspired from the GMSH syntax. 
+
+1. Basic shapes definitions. Shapes are defined with different parameters: shape ID, shape type, arguments, position, orientation.
+2. Boolean operations definitions. Operations are defined by: operation ID, operation type, shape 1 ID, shape 2 ID. 
+
+Shapes and boolean operations each have an ID, and boolean operations can reference previous operations to build upon them. 
+
+The ``impeller.composite`` file contains these instructions.
+
+.. warning:: 
+	The .composite file cannot contain comments at this point in time.
+
+.. code-block:: text
+
+  shapes
+  0; cylinder; 0.025:0.375; 0:0:0.125 ; 0:0:0
+  1; cylinder; 0.05:0.0416666;0:0:-0.208333 ; 0:0:0
+  2; rectangle; 0.083333333:0.00625:0.0416666;0.08333333:0:-0.208333 ; 0.7853981:0:0
+  3; rectangle; 0.00625:0.083333333:0.0416666;0:0.08333333:-0.208333; 0:0.7853981:0
+  4; rectangle; 0.083333333:0.00625:0.0416666;-0.08333333:0:-0.208333 ; -0.7853981:0:0
+  5; rectangle; 0.00625:0.083333333:0.0416666;0:-0.08333333:-0.208333 ; 0:-0.7853981:0
+  operations
+  6; union     ; 0:1
+  7; union     ; 6:2
+  8; union     ; 7:3
+  9; union     ; 8:4
+  10; union    ; 9:5
+  
+Let go over each line:
+
+* ``shapes``: introduces the shapes section.
+* ``0; cylinder; 0.025:0.375; 0:0:0.125 ; 0:0:0``: defines a cylinder, with ID equal to 0. Its radius is 0.025 and its half-length 0.375. Its z position is 0.125 and its orientation is default (aligned with Z axis).
+* The next shapes are defined in a similar fashion, and are assigned IDs 1, 2, 3, 4, and 5.
```

Comment:
Why not explaining the rectangle in the same way as you did with the cylinder? Clearly, the rectangle does not have a radius. Then, you can state that all other shapes are defined in the same fashion.

Reply:
Good point! I added the description for the 1st hyper rectangle, but left the rest as it is. Thanks!